tests/testapp/tests/sync/test_operations.py

```diff
@@ -86,6 +86,22 @@ def test_all_fsics(self):
         self.assertRecordsBuffered(self.data["group1_c2"])
         self.assertRecordsBuffered(self.data["group2_c1"])
 
+    def test_very_many_fsics(self):
+        """
+        Regression test against 'Expression tree is too large (maximum depth 1000)' error with many fsics
+        """
+        fsics = {self.data["group1_id"].id: 1, self.data["group2_id"].id: 1}
+        fsics.update({
+            uuid.uuid4().hex: i
+            for i in range(2000)
```

Comment:
Since we're packing it all back into a single query now, I wonder if it might be good to increase the size of this by an order of magnitude just to be sure we're not just kicking the can down the road?

Reply:
(Particularly as there may be maximum query lengths in SQLite)